### DISCUSSION TITLE
Best practice for realtime system, when disaster strikes and a consumer group is offline

### DISCUSSION THREAD

**Discussion by paulm17** [apache/pulsar#22236] [Q&A]
Hi all.

I have the following configuration for a geo-cluster:

![Replication](https://i.imgur.com/VPQHUSd.png)

A producer can submit messages and all the consumers will receive the messages.

The problem I am experiencing is the following:

If a region has no running consumers.  That region will not process any messages.  Should the consumer group come back up.  There are no messages to process as the other regions have acknowledged them and these acknowledgements have already been replicated to the local cluster.

I have thought about trying to track the message ids however, with a cluster, the message ids will go in a round robin configuration.  How will I know what is the minimum message to be processed?  If I was running just 1 subscriber, this would be easy to track and then have a receiver process rewind and then process all the messages up to that point.

Has anyone faced this issue and how did you resolve it?

Thanks

**Comment by david-streamlio** [comment]
Why must you re-process the messages in the region that experienced the outage? Assuming that the consumer groups in the other regions perform the same business logic, there shouldn't be a need to process them again.

**Reply by paulm17** [reply]
This is a real-time system for messages:

![Image Description](https://i.imgur.com/sM4XOgL.png)

How it works right now:

1) A user composes a message in a UI
2) The message gets sent over websockets to the websocket server (centrifugo): https://github.com/centrifugal/centrifugo
3) Internally, there is a pub/sub within centrifugo which will send the message to pulsar
4) The other consumers (in the other regions) will pick up the message with the consumer
5) The consumer will send the message to centrifugo websocket server
6) The websocket server will send the message to the connected clients
7) The UI will see the message

----

The problem that I am attempting to solve.  Is that clients connected in region East do not receive any websocket events from region West or Central (or any other variations).  There is no connection and won't ever be.  Centrifugo does not work in that way.

If a browser connects to East, they'll only see East websocket events.
If a browser connects to Central, they'll only see Central websocket events.
If a browser connects to West, they'll only see West websocket events.

Pulsar is what connects all the locations together.

In other words, subscribers have to process these messages for their respective region.  

----

Unfortunately I'm not able to use YugaByte CDC for this.  As I was thinking of using the sink.  But then fall short of needing another process to connect YB CDC with centrifugo and the other reasons.  It adds more complexity...
